File: nodecore-cli/src/main/java/nodecore/cli/serialization/TransactionReferenceInfo.java
// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.serialization;

import nodecore.api.grpc.VeriBlockMessages;
import nodecore.api.grpc.utilities.ByteStringUtility;

public class TransactionReferenceInfo {
    public TransactionReferenceInfo(final VeriBlockMessages.TransactionInfo transactionInfo) {
        blockNum = transactionInfo.getBlockNumber();
        confirmations = transactionInfo.getConfirmations();
        transaction = new TransactionInfo(transactionInfo.getTransaction());
        timestamp = transactionInfo.getTimestamp();
        endorsedBlockHash = ByteStringUtility.byteStringToHex(transactionInfo.getEndorsedBlockHash());
        bitcoinBlockHash = ByteStringUtility.byteStringToHex(transactionInfo.getBitcoinBlockHash());
        bitcoinTxID = ByteStringUtility.byteStringToHex(transactionInfo.getBitcoinTxId());
        bitcoinConfirmations = transactionInfo.getBitcoinConfirmations();
    }

    public int timestamp;

    public int blockNum;

    public int confirmations;

    public TransactionInfo transaction;

    public String endorsedBlockHash;

    public String bitcoinBlockHash;

    public String bitcoinTxID;

    public int bitcoinConfirmations;
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/TransactionReferenceInfo.kt
// VeriBlock NodeCore CLI
// Copyright 2017-2021 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import nodecore.api.grpc.RpcTransactionInfo
import nodecore.api.grpc.utilities.extensions.toHex

class TransactionReferenceInfo(
    transactionInfo: RpcTransactionInfo
) {
    val timestamp = transactionInfo.timestamp

    val blockNum = transactionInfo.blockNumber

    val confirmations = transactionInfo.confirmations

    val transaction = TransactionInfo(transactionInfo.transaction)

    val endorsedBlockHash = transactionInfo.endorsedBlockHash.toHex()

    val bitcoinBlockHash = transactionInfo.bitcoinBlockHash.toHex()

    val bitcoinTxID = transactionInfo.bitcoinTxId.toHex()

    val bitcoinConfirmations = transactionInfo.bitcoinConfirmations
}
